### PR TITLE
FEAT: improps and immeta

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
Closes: https://github.com/imageio/imageio/issues/362 https://github.com/imageio/imageio/issues/382 https://github.com/imageio/imageio/issues/501 https://github.com/imageio/imageio/issues/502 https://github.com/imageio/imageio/issues/503

This PR adds two new functions to the v3 API: `iio.improps` and `iio.immeta`. `improps` returns a ImageProperties object which contains standardized metadata, and `immeta` returns a dict object that contains format-specific metadata. 

It also adds a `imageio.core.v3_plugin_api.PluginV3` class that documents the plugin-side API for v3 plugins.

Further, I added implementations for `metadata` and `properties` to the v3 pillow plugin to support the newly added functions. For this plugin, reading of `properties` happens (as far as I know) without touching pixel data.

@almarklein @jni strike one ⚾ 🎉 

Still pending:

- [x] implement `properties` and `metadata` for legacy_plugin_wrapper (this will likely touch pixel data in the process)
- [x] implement `immeta` wrapper around `imopen`
- [x] `spacing` in `ImageProperties` (@almarklein where is it stored? Do you have an example image I can test with?)

### CODE REVIEW

**Comment by FirefoxMetzger** [comment]
blocked by #740

**Comment by almarklein** [comment]
> Do you have an example image I can test with?

The Dicom images specify it, for sure. The dicom plugin puts "sampling" in its metadata:
https://github.com/imageio/imageio/blob/e104505d4b5a53b11fd0023c83829795d4b31853/imageio/plugins/_dicom.py#L462-L464

**Comment by FirefoxMetzger** [comment]
> The Dicom images specify it, for sure. 

Nice. that sorts out one plugin. I was hoping for something that is loaded by pillow (and potentially one for ffmpeg and tifffile), because those seem to be our major plugins. At least if we go by the number of issues that we receive for each plugin.

**Comment by FirefoxMetzger** [comment]
@almarklein RTR.

I have decided to only add superficial support for wrapped legacy plugins at this stage. This means it won't crash if you call `immeta` or `improps` with legacy plugins, but it won't use the most efficient approach either. I will change this incrementally as I iterate over plugins to update them to v3.

File: imageio/core/v3_plugin_api.py
Comment (almarklein) [inline imageio/core/v3_plugin_api.py:29]:
```suggestion
        to that axis is a single float. If the spacing is non-uniform, the value
```

**Review by almarklein — APPROVED** [review]

**Comment by FirefoxMetzger** [comment]
@almarklein I take it that you approving the PR means you are happy with the changes otherwise and I can merge this after CI has passed?

**Comment by almarklein** [comment]
Yes!

**Pull request merged**
